### PR TITLE
Base64 master key

### PR DESCRIPTION
**Pull request by mdpye**

### CODE REVIEW

**Comment by mdpye** [comment]
It appears I have a little work to do to make it valid 2.7...

File: pusher/pusher_client.py [hunk 1]
```diff
@@ -36,16 +36,38 @@
...
-            backend, **backend_options)
+                app_id,
+                key,
+                secret,
+                ssl,
+                host,
+                port,
+                timeout,
+                cluster,
+                parsed_master_key,
```
Comment (damdo) [inline pusher/pusher_client.py]: I *think* here the `parsed_master_key` is setting the `encryption_master_key` argument no matter if an `encryption_master_key_base64` or `encryption_master_key` has been specified.

This means that the `json_encoder` var is setting the following argument, which is `encryption_master_key_base64` and not the `json_encoder` argument.

Comment (damdo) [reply pusher/pusher_client.py]: We should probably keep them both in the super constructor,
```suggestion
                encryption_master_key,
                encryption_master_key_base64,
``` 
check them below before the `host` checking and re-set via `self._`+`argname`.

Comment (mdpye) [reply pusher/pusher_client.py:66]: I suspect I don't understand what's going on here :/

**Comment by damdo** [comment]
LGTM 🎉

**Pull request merged**